pyrad/tools.py

```diff
@@ -17,15 +17,20 @@ def EncodeString(str):
         return str
 
 
-def EncodeOctets(str):
-    if len(str) > 253:
+def EncodeOctets(octetstring):
+    if len(octetstring) > 253:
         raise ValueError('Can only encode strings of <= 253 characters')
 
-    if str.startswith(b'0x'):
-        hexstring = str.split(b'0x')[1]
+    if type(octetstring) == bytes and octetstring.startswith(b'0x'):
```

Comment:
Python has 'isinstance' for it. Like this:
`if isinstance(octetstring, bytes) and octetstring.startswith(b'0x'):`